src/datasets/arrow_dataset.py

```diff
@@ -652,6 +656,7 @@ def load_from_disk(dataset_path: str, fs=None, keep_in_memory=False) -> "Dataset
         with open(Path(dataset_path, DATASET_INFO_FILENAME).as_posix(), "r", encoding="utf-8") as dataset_info_file:
             dataset_info = DatasetInfo.from_dict(json.load(dataset_info_file))
 
+        keep_in_memory = keep_in_memory if keep_in_memory is not None else is_small_dataset(dataset_info.dataset_size)
```

Comment:
Yes @lhoestq, you are right: dataset_size is always None when `load_from_disk` is called. However, the argument passed to `load_from_disk` is a directory (not an Arrow file) and it might contain multiple subdirectories (train, test), with multiple Arrow files. Should I make a scan and sum up all Arrow data file sizes?

Reply:
I think the `is_small_dataset` check can be done only in `Dataset.load_from_disk` (`datasets.load.load_from_disk` amd `DatasetDict.load_from_disk` can just pass the `keep_in_memory ` argument). So yes I think you just need to take the list of arrow files and sum their sizes.

In the code of `Dataset.load_from_disk` you can get the paths of the arrow files from `state["_data_files"]`.
For example the arrow files are used to load the table this way:

https://github.com/huggingface/datasets/blob/2a05294e2f8bc66983137797f213b325c78737b5/src/datasets/arrow_dataset.py#L654-L657

Reply:
Done @lhoestq. However, note that for the test I have implemented, the file size is 512 while once loaded, the dataset size in-memory is just 148.

Reply:
Yea I guess this is because of how pyarrow does memory allocation. The difference between the two values should stay close to a few KB.